sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data/DoubleExponentialHistogramBuckets.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.data;
+
+import com.google.auto.value.AutoValue;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * DoubleExponentialHistogramBuckets represents either the positive or negative measurements taken
+ * for a {@link DoubleExponentialHistogramPointData}.
+ */
+@AutoValue
+@Immutable
+public abstract class DoubleExponentialHistogramBuckets {
+  DoubleExponentialHistogramBuckets() {}
+
+  /**
+   * Create DoubleExponentialHistogramBuckets.
+   *
+   * @param offset Signed integer representing the bucket index of the first entry in bucketCounts.
```

Comment:
Can offset be negative? If so, what does that mean, exactly?

Reply:
Is "offset" here the exponent of the first bucket? So, `offset=-3` means that the first bucket is at `0.001` ?

If so, maybe `offset` isn't the best name for this parameter. `firstBucketExponent` maybe?

Reply:
There's also a "scale" that determines a portion of the exponent.  

This is really the "number of skipped buckets" because we expect the area around 0 to be uninhabited, so it's an optimization.

Specifically you have   `(base^(offset+i)` IIRC (and a negative number means you're in fractional starting buckets).

The exponent also has 2^scale = base.   Not sure this helps with naming (I'm bad at naming) but I'd be a fan of sticking to the otel proto naming unless we really really hate the name.

Reply:
The bucket boundaries are calculated via the `scale` and `offset`.

Firstly, `base = (2^(2^-scale))`, and then the bucket lower bound it calculated by `lower_bound = base^(i+offset)` like @jsuereth mentioned. The simplest case is when `scale = 0` and `offset=0`. That would mean your buckets are lower bounded at 1.0, 2.0, 4.0, 8.0, 16.0, etc. With `offset = -3` your buckets are lower bounded at 0.125, 0.25, 0.5, 1.0, 2.0, 4.0, 16.0, etc.

I think it makes sense to stick to the `offset` name since that's what it's called in the protocol. In terms of the javadoc description, I based it off the [protocol's description](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L535), but it could probably be clearer.

How does this sound instead? `@param offset Signed integer which shifts the bucket boundaries according to `lower_bound = base^(offset+i)`.

Reply:
If you could put all that text that you wrote into the class level javadoc, that would be best. I don't want us to have to somehow depend on the specs or the proto definitions to understand the code here.

Reply:
Alright, I've: 
- Put more detail on the class level javadocs, with examples of how bucket boundaries are calculated.  
- Put more detail on the `@param scale` and altered the `@param offset` javadoc to reflect my last comment.  
- Added javadoc for `getScale()` and `getOffset()`

Reply:
Thanks! Very helpful for those of us who aren't intimately familiar with the art.